interpreter/cling/lib/Interpreter/IncrementalParser.cpp

```diff
@@ -392,6 +392,28 @@ namespace cling {
            && (!initialized || (m_TransactionPool && m_Parser));
   }
 
+  namespace {
+    template <class T>
+    struct Reverse {
+      const T &m_orig;
+      auto begin() -> decltype(m_orig.rbegin()) { return m_orig.rbegin(); }
+      auto end() -> decltype (m_orig.rend()) { return m_orig.rend(); }
+    };
+    template <class T>
+    Reverse<T> revert(const T& orig) { return {orig}; }
```

Comment:
Can you use the spelling 'reverse'  (which is the standard's spelling: https://en.cppreference.com/w/cpp/algorithm/reverse)?  When first reading that I understood 'revert' as in 'undo' rather than change the direction of the iteration.